### DISCUSSION TITLE
Bufferbloat, dscp, and Timestamping packets

### DISCUSSION THREAD

**Discussion by dtaht** [fonoster/routr#202] [Ideas]
I am of course a big fan of fq_codel or cake everywhere, especially for voip scenarios.  I have longed to find a product that provided a bit of insight as to what was going wrong when voip was in use.

This includes tracking dscp survivability (are you sending and getting dscp marked packets (EF, VA), from the endpoint? While a lot of big providers mangle or bleach these, most of the small ones don´t (such as starlink and most wisps) and it would be good to track in the server (via the cmsg facility on a per packet basis).

Similarly, via cmsg, you can pull per packet timestamps and see bursts or jitter that way, coming from the endpoint. 

Huge fan of telling voip users to make sure fq_codel or cake were also on the path, and looking at backlogs and drops there.

**Comment by psanders** [comment]
I've moved this to discussions until I get a better idea of how we can best incorporate it into the stack.
